web/src/main/kotlin/no/nav/su/se/bakover/web/routes/sak/SakRoutes.kt

```diff
@@ -37,22 +38,29 @@ internal fun Route.sakRoutes(
     sakService: SakService
 ) {
     get(sakPath) {
-        call.lesFnr("fnr").fold(
-            ifLeft = { call.svar(BadRequest.message(it)) },
-            ifRight = { fnr ->
+        call.lesFnr("fnr")
+            .mapLeft { BadRequest.message(it) }
+            .flatMap { fnr ->
+                call.suUserContext.assertBrukerHarTilgangTilPerson(fnr)
+
                 sakService.hentSak(fnr)
-                    .mapLeft { call.svar(NotFound.message("Fant ikke noen sak for person: $fnr")) }
+                    .mapLeft { NotFound.message("Fant ikke noen sak for person: $fnr") }
                     .map {
                         call.audit("Hentet sak for fnr: $fnr")
-                        call.svar(Resultat.json(OK, serialize((it.toJson()))))
+                        Resultat.json(OK, serialize((it.toJson())))
                     }
             }
-        )
+            .fold(
+                ifLeft = { call.svar(it) },
+                ifRight = { call.svar(it) }
+            )
     }
 
     get("$sakPath/{sakId}") {
-        call.withSak(sakService) {
-            call.svar(Resultat.json(OK, serialize((it.toJson()))))
+        call.withSak(sakService) { sak ->
+            call.suUserContext.assertBrukerHarTilgangTilPerson(sak.fnr)
```

Comment:
Hva gjør vi for endepunkter som ikke "tar inn fnr"? Vi burde vel i utgangspunktet hatt den samme sjekken for alle endepunktene?

Reply:
Akkurat dette endepunktet tar ikke inn fnr, så da må vi gjøre som her; få tak i noe som gir oss fnr, og så gjør sjekken på det.

Reply:
Hvis vi bruker sakid som en "base" i path hver gang er det kanskje mulig å generalisere dette slik at vi kan sjekke tilgang  for alle kall? Bør nok få på noe cacheing et sted isåfall, men kunne kanskje vært en mulighet?

Reply:
Vi har vel allerede `sakId` i de fleste pathene våre som går på saksbehandling iallfall, så det er kanskje mulig å få til. Kan se om jeg får til noe som funker og ikke skjuler/roter til ting for mye.

Reply:
Kunne vi i den SuUserContext, gjøre som Jacob sier, gjøre denne sjekken. f.eks. 
```
when{path} {
"/saker/" -> finn sakId, hent fnr via sakId, etc.
"custom path 1 med fnr" -> slå opp direkte på fnr
"custom path 2 med fnr" -> samme
else -> throw SomeException

```

Reply:
Også bør vi sende med f.eks. User/Person (som bør ligge i domene-laget), inn til Servicen, slik at vi slipper å gjøre nye kall til PDL

Reply:
Det hadde sikkert funket ja, men jeg syns det virker veldig error-prone. Hvis det bare ligger og durer i bakgrunnen og tilfeldigvis plukker opp paths som vi bruker så ser jeg for meg at det er veldig fort gjort å glemme å legge til disse sjekkene når man legger til nye endepunkter.

En annen ting er at hvis den bare gjør ting internt der så får vi ikke på noen god måte returnert/gitt f.eks. person eller sak til rute-handleren, noe som fører til at vi kan komme til å duplisere kall både til PDL og egne servicer/database.